src/app/utils.ts

```diff
@@ -96,5 +97,30 @@ export function getWindowUrl(window: typeof windows[]) {
 		throw new Error(`Invalid window name: ${window}`)
 	}
 
-	return `${APP_ORIGIN}/talk_desktop__window_${window}/index.html`
+	const origin = appData.serverUrl ? new URL(appData.serverUrl).origin : APP_ORIGIN
+	return `${origin}/talk_desktop__window_${window}/index.html`
+}
+
+/**
+ * Check weather a link is an internal application link
+ *
+ * @param url - URL
+ */
+export function isInternalUrl(url: string | URL) {
+	if (typeof url === 'string') {
+		url = new URL(url)
+	}
+
+	const internalOrigin = appData.serverUrl ? new URL(appData.serverUrl).origin : APP_ORIGIN
```

Comment:
can it be moved out of function calls (here and at L100) and reused?

Reply:
In a simple way - no. The value can be changed in runtime, e.g., you logged out and logged in again. So it at least must be in the app data.

Will refactor it much when we add Multi-Account Support